src/cryptonote_core/blockchain.cpp

```diff
@@ -3228,51 +3225,77 @@ bool Blockchain::check_tx_inputs(transaction& tx, tx_verification_context &tvc,
       }
 
       crypto::public_key const &state_change_service_node_pubkey = quorum->workers[state_change.service_node_index];
-      const uint64_t height                                      = state_change.block_height;
-      constexpr size_t num_blocks_to_check                       = service_nodes::STATE_CHANGE_TX_LIFETIME_IN_BLOCKS;
-
-      std::vector<std::pair<cryptonote::blobdata,block>> blocks;
-      std::vector<cryptonote::blobdata> txs;
-      if (!get_blocks(height, num_blocks_to_check, blocks, txs))
+      if (hf_version >= cryptonote::network_version_12_checkpointing)
       {
-        MERROR_VER("Failed to get historical blocks to check against previous state changes for de-duplication");
-        return false;
-      }
+        //
+        // NOTE: Query the Service Node List for the in question Service Node the state change is for and disallow if conflicting
+        //
+        std::vector<service_nodes::service_node_pubkey_info> service_node_array = m_service_node_list.get_service_node_list_state({state_change_service_node_pubkey});
+        if (service_node_array.empty())
+        {
+          MERROR_VER("Service Node no longer exists on the network, state change can be ignored");
```

Comment:
Is this actually an error?  Seems like INFO/WARNING

Reply:
LOG_PRINT_L2'ed.